compaction/compaction_manager.cc

```diff
@@ -565,7 +565,7 @@ sstables::shared_sstable compaction_manager::sstables_task::consume_sstable() {
 }
 
 future<semaphore_units<named_semaphore_exception_factory>> compaction_manager::task::acquire_semaphore(named_semaphore& sem, size_t units) {
-    return seastar::get_units(sem, units, _compaction_data.abort).handle_exception_type([this] (const semaphore_aborted& e) {
+    return seastar::get_units(sem, units, _compaction_data.abort).handle_exception_type([this] (const abort_requested_exception& e) {
```

Comment:
From #10666 it seemed to me that the plan was to re-derive `named_semaphore_aborted` from `semaphore_aborted`

Reply:
Yes, but we can do both, belts and suspenders.

Reply:
This change implements a generic approach, while the seastar change can help future use cases.